components/script/script_thread.rs

```diff
@@ -1969,21 +1963,22 @@ impl ScriptThread {
     /// https://html.spec.whatwg.org/multipage/#navigating-across-documents
     /// The entry point for content to notify that a new load has been requested
     /// for the given pipeline (specifically the "navigate" algorithm).
-    fn handle_navigate(&self, pipeline_id: PipelineId, subpage_id: Option<SubpageId>, load_data: LoadData) {
+    fn handle_navigate(&self, parent_pipeline_id: PipelineId, pipeline_id: Option<PipelineId>, load_data: LoadData) {
         // Step 8.
         {
             let nurl = &load_data.url;
             if let Some(fragment) = nurl.fragment() {
-                let document = match self.root_browsing_context().find(pipeline_id) {
+                let document = match self.root_browsing_context().find(parent_pipeline_id) {
                     Some(browsing_context) => browsing_context.active_document(),
-                    None => return warn!("Message sent to closed pipeline {}.", pipeline_id),
+                    None => return warn!("Message sent to closed pipeline {}.", parent_pipeline_id),
                 };
+                let document = document.r();
```

Comment:
Note: @asajeffrey removed this line in #11695, but this seemed inadvertent to me so I brought it back. Not sure about this change.